### DISCUSSION TITLE
Use custom domain for Google Login

### DISCUSSION THREAD

**Discussion by Kahtaf** [Web3Auth/Web3Auth#510] [Help]
How can we change the domain from `openlogin.com` to a custom domain? Is there a way we can host the functionality that `beta.openlogin.com` provides so we can use our own domain instead?

![image](https://user-images.githubusercontent.com/5796181/187296190-ddfad274-d93f-4eec-a186-8d723f71a70d.png)

**Comment by yashovardhan** [comment]
Hey @Kahtaf,

Thanks for your questions. `Openlogin` is a part of Web3Auth Plug and Play which implements the key reconstruction and enables you to easily integrate with a couple of lines of code. This particular step cannot be skipped in the Web3Auth Plug and Play since the key reconstruction needs to be done in the frontend to make sure there's no networks calls with the full key anywhere and maintain the non custodiality of Web3Auth.

That being said, there are ways you can still use web3auth without the openlogin aspect. This is enabled by our Web3Auth Self Host, which basically allows you to implement your own Openlogin logic directly within your custom domain and use it accordingly. You can read more about it here: https://web3auth.io/docs/self-hosting

**Comment by Kahtaf** [comment]
Thanks for your reply @yashovardhan. I found that self-hosting was **not** the only way we could use our own branding on Google/Twitter/etc. 

Suppose we use custom authentication and a federated identity provider (Auth0, Firebase, etc) and use the Web3Auth core login flow. In that case, we can customize the Google login window with our own domain rather than `openlogin.com`. 

The flow looks like this: 
1. Login button on our app
2. Federated identity provider login screen (ex: Auth0)
3. Social app login screen (ex: Google) with our own branding
4. Web3Auth window to construct the key
5. Return to our app with the user signed in

**Reply by yashovardhan** [reply]
Hey Kahtaf,

To clarify your answer here, the key reconstruction is still happening on openlogin.com. You're just using your own custom authentication and providing the jwt token to web3auth.

If you see the flow in the auth0 guide, you'll get a better understanding of how it works.

https://web3auth.io/docs/guides/auth0

I mentioned self host since your question was to change the domain of openlogin.com, which isn't possible without web3auth self host.
